### PR TITLE
Set python_shell=True to use pipes and other shell features

### LINKED ISSUE #118: systemd detection command throws "garbage option" error
Applying the docker.containers SLS on CentOS 7 results in the following errors at render time:

```
[INFO    ] Executing command '/usr/bin/ps -p1 | grep -q systemd && echo systemd || echo upstart' in directory '/root'
[ERROR   ] Command '/usr/bin/ps -p1 | grep -q systemd && echo systemd || echo upstart' failed with return code: 1
[ERROR   ] output: error: garbage option

Usage:
 ps [options]

 Try 'ps --help <simple|list|output|threads|misc|all>'
  or 'ps --help <s|l|o|t|m|a>'
 for additional help text.

For more details see ps(1).
```

This is due to changes in the behavior of `cmd.run` in Salt 2017.7.0:
https://docs.saltstack.com/en/latest/topics/releases/2017.7.0.html#general-deprecations

### PR DESCRIPTION
**Pull request by xenophonf**
Fixes #118

### CODE REVIEW

File: docker/containers.sls [hunk 1]
```diff
@@ -11,7 +11,7 @@ docker-image-{{ name }}:
       - service: docker-service
 
 {# TODO: SysV init script #}
-{%- set init_system = salt["cmd.run"]("ps -p1 | grep -q systemd && echo systemd || echo upstart") %}
+{%- set init_system = salt["cmd.run"]("ps -p1 | grep -q systemd && echo systemd || echo upstart", python_shell=True) %}
```
Comment (vutny) [inline docker/containers.sls]: Modern Salt versions provide a couple of nice alternatives to this:
```jinja
{%- set init_system = grains['init'] %}
{# or #}
{%- set init_system = salt['test.provider']('service') %}
```

Comment (xenophonf) [reply docker/containers.sls]: This change fixes the immediate problem.  I'd be happy to submit another PR later that refactors the code in question.

Comment (vutny) [reply docker/containers.sls]: The Clean Code defines so-called "The Boy Scout Rule": Leave the campground cleaner than you found it.
If something could be done in the right way, it should be immediately done so with no intermediate steps.
But that's up to you and the formula maintainers, of course. This is only my opinion and it's good to make any valuable contribution. Thanks.

**Comment by blbradley** [comment]
I agree with @vutny. You should be using data from grains to check which init system is in place.

**Comment by blbradley** [comment]
Fixed by #122

**Pull request closed without merging**